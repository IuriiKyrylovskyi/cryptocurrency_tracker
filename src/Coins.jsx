import React from 'react';
import Coin from './Coin';
import { AiOutlineReload } from 'react-icons/ai';
import PropTypes from 'prop-types';

const Coins = ({ coins }) => {
  if (coins.length === 0) {
    return (
      <div className="load">
        <AiOutlineReload className="load-icon" />
        <p>Loading...</p>
      </div>
    );
  }
  return (
    <div className="coins">
      <div className="coins__row">
        <h3 className="coins__coin">Coin</h3>
        <h3 className="coins__volume">Volume</h3>
        <h3 className="coins__price">Price</h3>
        <h3 className="coins__price-change24">24h change</h3>
      </div>
      {coins.map(coin => (
        <Coin key={coin.id} coin={coin} />
      ))}
    </div>
  );
};

Coins.propTypes = {
  coins: PropTypes.array,
};

export default Coins;
