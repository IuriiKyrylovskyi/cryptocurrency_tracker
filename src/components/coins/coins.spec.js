import React from 'react';
import { shallow } from 'enzyme';
// import { fetchCryptoList } from '../../gateway';
import Coins from './Coins';

const setUp = props => shallow(<Coins {...props} />);

const props = [
  {
    id: 'bitcoin',
    symbol: 'btc',
    name: 'Bitcoin',
    current_price: 41593,
    image: 'https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579',
    market_cap: 778357915130,
    market_cap_change_percentage_24h: -4.57821,
    max_supply: 21000000,
    price_change_24h: -1971.47108004,
    price_change_percentage_24h: -4.52542,
    total_supply: 21000000,
    total_volume: 141884973661,
  },
  {
    id: 'dogecoin',
    symbol: 'dgc',
    name: 'dogecoin',
    current_price: 415930,
    image: 'https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579',
    market_cap: 778357915130,
    market_cap_change_percentage_24h: -4.57821,
    max_supply: 21000000,
    price_change_24h: -1971.47108004,
    price_change_percentage_24h: -4.52542,
    total_supply: 21000000,
    total_volume: 141884973661,
  },
];

let component;

// const mockJsonPromise = Promise.resolve({ coins: fetchCryptoList });
// const mockFetchPromise = Promise.resolve({ json: () => mockJsonPromise });
// global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);

describe('Coins component', () => {
  describe('should render Coins component', () => {
    beforeEach(() => {
      component = setUp();
    });

    it('should render component correctly', () => {
      expect(component).toMatchSnapshot();
    });
    // it('should call fetch in useEffect', () => {
    //   expect(global.fetch).toMatchSnapshot();
    // });

    // it('should contain .coins wrapper', () => {
    //   const wrapper = component.find('.coins');

    //   expect(wrapper.length).toBe(1);
    // });
  });

  describe('should render Coins component with props', () => {
    it('should render .coins__list with 2 items', () => {
      component = setUp(props);
      const list = component.find('.coins__list');
      expect(list).toHaveLength(2);
    });
  });

  describe('should render empty list', () => {
    it('should render .coins__list with 0 items', () => {
      component = setUp([]);
      const list = component.find('.coins__list');
      expect(list.length).toBe(0);
    });
  });
});
