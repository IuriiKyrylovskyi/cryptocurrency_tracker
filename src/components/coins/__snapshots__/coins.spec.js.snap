// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coins component should render Coins component should render component correctly 1`] = `
<div
  className="load"
>
  <AiOutlineReload
    className="load-icon"
  />
  <p>
    Loading...
  </p>
</div>
`;

exports[`Coins component should render Coins component with props should render .coins__list with 2 items 1`] = `
<div
  className="coins"
>
  <div
    className="coins__row"
  >
    <h3
      className="coins__coin"
    >
      Coin
    </h3>
    <h3
      className="coins__volume"
    >
      Volume
    </h3>
    <h3
      className="coins__price"
    >
      Price
    </h3>
    <h3
      className="coins__price-change24"
    >
      24h change
    </h3>
  </div>
  <ul
    className="coins__list"
  >
    <Coin
      coin={
        Object {
          "current_price": 41593,
          "id": "bitcoin",
          "image": "https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579",
          "market_cap": 778357915130,
          "market_cap_change_percentage_24h": -4.57821,
          "max_supply": 21000000,
          "name": "Bitcoin",
          "price_change_24h": -1971.47108004,
          "price_change_percentage_24h": -4.52542,
          "symbol": "btc",
          "total_supply": 21000000,
          "total_volume": 141884973661,
        }
      }
      current_price={100}
      image="coin image"
      key="bitcoin"
      name="coin name"
      price_change_percentage_24h={0}
      symbol=""
    />
    <Coin
      coin={
        Object {
          "current_price": 415930,
          "id": "dogecoin",
          "image": "https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579",
          "market_cap": 778357915130,
          "market_cap_change_percentage_24h": -4.57821,
          "max_supply": 21000000,
          "name": "dogecoin",
          "price_change_24h": -1971.47108004,
          "price_change_percentage_24h": -4.52542,
          "symbol": "dgc",
          "total_supply": 21000000,
          "total_volume": 141884973661,
        }
      }
      current_price={100}
      image="coin image"
      key="dogecoin"
      name="coin name"
      price_change_percentage_24h={0}
      symbol=""
    />
  </ul>
</div>
`;
